packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.classNames.ts

```diff
@@ -0,0 +1,76 @@
+import { memoizeFunction } from '../../Utilities';
+import {
+  ITheme,
+  IPalette,
+  mergeStyleSets,
+  getFocusStyle
+} from '../../Styling';
+
+export interface ISwatchColorClassName {
+  themeClass?: string | undefined;
+}
+
+export const getClassNames = memoizeFunction((
+  theme: ITheme,
+  color: string
+): ISwatchColorClassName => {
+  const themedColor: string | undefined = getSwatchThemeColor(theme, color);
+
+  return mergeStyleSets({
+    themeClass: [themedColor ?
+      {
+        fill: themedColor
+      } :
+      undefined
+    ]
+  });
+
+});
+
+function getSwatchThemeColor(theme: ITheme, color: string) {
```

Comment:
Again, the color picker doesn't need to know about the theme. Your app should just be passing in color codes. When the theme changes, just rerender the component with the new color values.

Reply:
Don't know how that works. If I do fill: color, how would that work?

Reply:
You don't need to add any code to swatch color picker to get it to render a theme color, it just works. See this [code pen](https://codepen.io/anon/pen/xXpQrg). If you want it to update when the theme changes, just rerender the component.

Reply:
Then again. this is not what my issue is. I know it does but then if I change the color of my site, the primary color won't change with theme.

Reply:
It will change if you rerender the component. Every time the component renders, it get the value of the theme. It seems like your problem is that your site isn't rerendering the color picker when the swatch value changes.

Reply:
Here's an example in a [codepen](https://codepen.io/anon/pen/JrMVJZ). Clicking the button updates the theme and rerenders the swatch picker. Obviously, you wouldn't do this with a force update in real code, but it shows the idea.